core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -24,6 +24,15 @@
 #include <cstdint>
 #include <string>
 
+// Use unlikely attribute from C++20 to improve performance of
+// reference counting when copy constructors are involved.
+#if defined(KOKKOS_ENABLE_IMPL_REF_COUNT_BRANCH_UNLIKELY) && \
+    defined(KOKKOS_ENABLE_CXX20)
```

Comment:
Should this be `!defined(KOKKOS_ENABLE_CXX17)`? If you use C++23, `KOKKOS_ENABLE_CXX20` won't be defined.

Reply:
Right. I will fix it. Alternatively is there a way to say something like `CXX_VERSION >= 20` ?

Reply:
No, `#ifndef KOKKOS_ENABLE_CXX17` is the easiest way right now.